src/Microsoft.IdentityModel.Validators/AadIssuerValidator/AadIssuerValidator.cs

```diff
@@ -390,14 +389,62 @@
 
             if (validIssuerTemplate.Contains(TenantIdTemplate))
             {
-                return validIssuerTemplate.Replace(TenantIdTemplate, tenantId) == actualIssuer;
+                return IssuersWithTemplatesAreEqual(validIssuerTemplate, TenantIdTemplate, actualIssuer, tenantId);
             }
             else
             {
```

Comment:
>issuer1 [](http://example.com/codeflow?start=65&length=7)

These are the values passed: signingKeyIssuer, tenantIdFromToken, tokenIssuer, tenantIdFromToken

The term template here is a bit confusing, what you call issuer1Template here really is the value that will be substituted on the template, why not call it what it is, the tenant Id, the same goes for the last parameter.

Reply:
Great, done